core/controllers/domain_objects_validator.py

```diff
@@ -83,27 +101,46 @@ def validate_change_dict_for_blog_post(change_dict):
         if not all(tag in list_of_default_tags for tag in change_dict['tags']):
             raise Exception(
                 'Invalid tags provided. Tags not in default tags list.')
+    return change_dict
 
 
-def validate_collection_change(obj):
+def validate_collection_change(collection_change_dict):
     """Validates collection change.
 
     Args:
-        obj: dict. Data that needs to be validated.
+        collection_change_dict: dict. Data that needs to be validated.
+
+    Returns:
+        dict. Returns collection change dict after validation.
     """
     # No explicit call to validate_dict method is necessary, because
     # CollectionChange calls validate method while initialization.
-    collection_domain.CollectionChange(obj) # type: ignore[no-untyped-call]
+
+    # Object should not be returned from here because it requires modification
+    # in many of the methods in the domain layer of the codebase and we are
+    # planning to remove collections from our codebase hence modification is
+    # not done here.
+
+    collection_domain.CollectionChange(collection_change_dict)
+    return collection_change_dict
```

Comment:
```suggestion
    # Object should not be returned from here because it requires modification
    # in many of the methods in the domain layer of the codebase and we are
    # planning to remove collections from our codebase hence modification is
    # not done here.
    collection_domain.CollectionChange(collection_change_dict)
    return collection_change_dict
```

Reply:
Updated.